### ISSUE TITLE
Protobuf issues in TensorFlow 2.12

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#52]
Due to some dependency update in `tensorflow-datasets` and where `tfds` does not have strict enough versioning, `protobuf` seems to be broken.

I managed to fix it for `tensorflow < 2.12` by force-reinstalling `protobuf` with a specific version like so:
```
pip install "protobuf<=3.20" --force-reinstall
```

However, for whatever reason, this does not seem to work for the latest `tensorflow` version.

This was observed on `ubuntu-20.04` with `Python 3.9`.

**Comment by andreped** [comment]
Disabled `tf==2.12` CI tests temporarily in 3b6d94479d5e923742cff2ef14c7b25d8156781b

**Comment by andreped** [comment]
Using older protobuf resolves issue for older tf versions, but for latest it might be that we need the latest version.
```
tensorflow 2.12.0 requires protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3, but you have protobuf 3.20.0 which is incompatible.
tensorflow-metadata 1.13.1 requires protobuf<5,>=3.20.3, but you have protobuf 3.20.0 which is incompatible.
```

**Comment by andreped** [comment]
Fixed in 0bc538c4f61e9ff10a3b844b26d71f6fc1e068e2.
